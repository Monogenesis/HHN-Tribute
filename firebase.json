{
  "hosting": {
    "public": "build",
    "ignore": [
      "firebase.json",
      "**/.*",
      "**/node_modules/**"
    ],
    "rewrites": [
      {
        "source": "**",
        "destination": "/index.html"
      }
    ]
  },
  "firestore": {},
  "functions": {
    "source": "functions"
  }
}
