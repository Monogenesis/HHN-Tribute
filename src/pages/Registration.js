import React from "react";
import "./pagestyles/LiveMap.css";
import { GatherGender } from "../components/GatherBaseData/GatherBaseData";
import { SubmitData } from "../components/GatherBaseData/GatherBaseData";
import "./pagestyles/Registration.css";
import { IconButton, Tooltip } from "@material-ui/core";
import { FcInfo } from "react-icons/fc";
const Registration = (props) => {
  return (
    <main>
      <section id="Header">
        <h1 className="RegistrationTitle">Registrierung</h1>
        <h2 className="RegistrationHeader">
          Bitte spenden Sie hier Ihre Daten
          <Tooltip
            title={
              <span>
                <h3>Hier geben Sie ihre Stammdaten an.</h3>
                <br />
                <h3>
                  Alter, Größe und Gewicht werden nur in fünfer Schritten
                  angegben, um ihre Angaben weiter zu anonymisieren.
                </h3>
                <br />
                <h3>
                  Bitte geben Sie mindestens ihre Postleitzahl an damit wir Ihre
                  Daten nutzen können.
                </h3>
              </span>
            }
            placement="bottom-start"
            enterDelay={0}
            enterTouchDelay={50}
            leaveTouchDelay={30000}
            arrow="true"
          >
            <IconButton>
              <FcInfo size={50} />
            </IconButton>
          </Tooltip>
        </h2>
        <p className="Space"></p>
      </section>

      <section className="tibutField">
        <table className="tributTable">
          <tbody>
            <tr>
              <td className="genderLabel">Geschlecht:</td>
              <td className="genderArea">
                <select className="dropdownGender" id="gender">
                  {props.GatherGender}
                </select>
              </td>
            </tr>
            <tr>
              <td className="ageLabel">Alter:</td>
              <td className="ageArea">
                <select className="dropdownAge" id="age"></select>
              </td>
            </tr>
            <tr>
              <td className="heightLabel">Größe:</td>
              <td className="heightArea">
                <select className="dropdownHeight" id="height"></select>
              </td>
            </tr>
            <tr>
              <td className="weightLabel">Gewicht:</td>
              <td className="weightArea">
                <select className="dropdownWeight" id="weight"></select>
              </td>
            </tr>
            <tr>
              <td className="zipCodeLabel">Postleitzahl:</td>
              <td className="zipCodeArea">
                <input className="fieldZipCode" id="zipCode"></input>
              </td>
            </tr>
            <tr>
              <td className="spaceHolder"></td>
              <td className="submitButtonField">
                {" "}
                <button type="button" id="submitButton" onClick={SubmitData}>
                  Daten spenden
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </section>
    </main>
  );
};

export default Registration;
