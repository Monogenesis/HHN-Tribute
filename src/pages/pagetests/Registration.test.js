import firebase from "../../Firebase.js" //Wird benÃ¶tigt 
import React from "react";
import renderer from "react-test-renderer";
import Registration from "../Registration";

test("Registration Page renders correctly", () => {
    const tree = renderer.create(<Registration></Registration>).toJSON();
expect(tree).toMatchSnapshot();
})