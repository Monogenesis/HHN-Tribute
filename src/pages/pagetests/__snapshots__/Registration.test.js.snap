// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Page renders correctly 1`] = `
<main>
  <section
    id="Header"
  >
    <h1
      className="RegistrationTitle"
    >
      Registration
    </h1>
    <h2
      className="RegistrationHeader"
    >
      Please enter your data here
    </h2>
    <p
      className="Space"
    />
  </section>
  <section
    className="tibutField"
  >
    <table
      className="tributTable"
    >
      <tbody>
        <tr>
          <td
            className="genderLabel"
          >
            Gender:
          </td>
          <td
            className="genderArea"
          >
            <select
              className="dropdownGender"
              id="gender"
            />
          </td>
        </tr>
        <tr>
          <td
            className="ageLabel"
          >
            Age:
          </td>
          <td
            className="ageArea"
          >
            <select
              className="dropdownAge"
              id="age"
            />
          </td>
        </tr>
        <tr>
          <td
            className="heightLabel"
          >
            Height:
          </td>
          <td
            className="heightArea"
          >
            <select
              className="dropdownHeight"
              id="height"
            />
          </td>
        </tr>
        <tr>
          <td
            className="weightLabel"
          >
            Weight:
          </td>
          <td
            className="weightArea"
          >
             
            <select
              className="dropdownWeight"
              id="weight"
            />
          </td>
        </tr>
        <tr>
          <td
            className="zipCodeLabel"
          >
            Zip Code:
          </td>
          <td
            className="zipCodeArea"
          >
            <input
              className="fieldZipCode"
              id="zipCode"
            />
          </td>
        </tr>
        <tr>
          <td
            className="spaceHolder"
          />
          <td
            className="submitButtonField"
          >
             
            <button
              id="submitButton"
              onClick={[Function]}
              type="button"
            >
              Submit
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </section>
</main>
`;
